src/Generator/GiftCardCodeGenerator.php

```diff
@@ -34,14 +31,14 @@ public function generate(): string
             /** @psalm-suppress ArgumentTypeCoercion */
             $code = bin2hex(random_bytes($this->codeLength));
             $code = preg_replace('/[01]/', '', $code); // remove hard to read characters
-            $code = mb_strtoupper(mb_substr($code, 0, $this->codeLength));
+            $code = mb_strtoupper(mb_substr((string) $code, 0, $this->codeLength));
```

Comment:
Asserting to check for string can avoid some issue in case the code is not already a string